### PR TITLE
Replace @octokit/rest with octokit in scaffolder-backend

### PR DESCRIPTION
**Pull request by Crevil**
As disucssed in #8263 the used package `@octokig/rest` is deprecated (see https://github.com/octokit/octokit.js/discussions/620#discussioncomment-917159 for details).

To get the ball rolling on fixing the mentioned issue this change migrates over to the new package.

Signed-off-by: Crevil <bjoern.soerensen@gmail.com>

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/grumpy-teachers-remain.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-scaffolder-backend': minor
```
Comment (jhaals) [inline .changeset/grumpy-teachers-remain.md:2]: There are no breaking changes in this PR right? In that case it should be `patch`

**Review by jhaals — COMMENTED** [review]
Looks good to me! Small NIT on the changeset.

As this is a new package did you do some basic test running this to see that at least some code path worked? :)

**Comment by benjdlambert** [comment]
Just tested this - seems to work well with github apps for now too, so nothing new needed which is great. Let's ship this once the `changeset` is updated. Other thing to note is I'm gonna create a ticket to improve the logging around `isomorphic-git` it would appear that pretty useful error messages that get returned from git providers in messages are ignored in favour of stuff like 403 permission denied, and the extra data is on the `error.data` which we currently don't propgate to the user.

**Comment by benjdlambert** [comment]
Here's that particular issue: https://github.com/backstage/backstage/issues/9208

File: .changeset/grumpy-teachers-remain.md (hunk 1 above)
Comment (Crevil) [reply .changeset/grumpy-teachers-remain.md]: It shouldn't brake anything 🙏 👍 Updated to a patch instead.

**Review by jhaals — APPROVED** [review]
Nice work @Crevil! 🚀

**Pull request merged**